pep-lint.py

```diff
@@ -0,0 +1,426 @@
+        else:
+            post_date, post_url = post[1:-4].split(" <")
+            yield from _date(line_num, post_date, "Post-History")
+            yield from _thread(line_num, post_url, "Post-History")
+
+
+def _validate_resolution(line_num, line):
+    """'Resolution' must be a direct thread/message URL"""
+
+    yield from _thread(line_num, line, "Resolution", allow_message=True)
+
+
+########################
+#  Validation Helpers  #
+########################
+
+def _pep_num(line_num, pep_number, prefix):
+    if pep_number.startswith("0") and pep_number != "0":
+        yield line_num, f"{prefix} must not contain leading zeros: {pep_number}"
+    if not _is_digits(pep_number):
+        yield line_num, f"{prefix} must be numeric: {pep_number}"
+    elif not 0 <= int(pep_number) <= 9999:
+        yield line_num, f"{prefix} must be between 0 and 9999: {pep_number}"
+
+
+def _is_digits(string):
+    return all(c in "0123456789" for c in string)
+
+
+def _email(line_num, author_email, prefix):
+    if EMAIL_PATTERN.match(author_email.strip()) is None:
+        yield line_num, f"{prefix} must be a list of 'Name <email@example.com>' entries: {author_email}"
+
+
+def _thread(line_num, url, prefix, allow_message=False, discussions_to=False):
+    msg = f"{prefix} must be a valid thread URL"
+
+    if not url.startswith("https://"):
+        if not discussions_to:
+            yield line_num, msg
+        return
+
+    if url.startswith("https://discuss.python.org/t/"):
+        # Discussions-To links must be the thread itself, not a post
+        pattern = DISCOURSE_THREAD_PATTERN if discussions_to else DISCOURSE_POST_PATTERN
+
+        remainder = url.removeprefix("https://discuss.python.org/t/").removesuffix("/")
+        if pattern.fullmatch(remainder) is not None:
+            return
+
+    if url.startswith("https://mail.python.org/pipermail/"):
+        remainder = url.removeprefix("https://mail.python.org/pipermail/")
+        if MAILMAN_2_PATTERN.fullmatch(remainder) is not None:
+            return
+
+    if url.startswith("https://mail.python.org/archives/list/"):
+        remainder = url.removeprefix("https://mail.python.org/archives/list/")
+        if allow_message and MAILMAN_3_MESSAGE_PATTERN.fullmatch(remainder) is not None:
+            return
+        if MAILMAN_3_THREAD_PATTERN.fullmatch(remainder) is not None:
+            return
+
+    yield line_num, msg
+
+
+def _date(line_num, date_str, prefix):
+    try:
+        dt = datetime.datetime.strptime(date_str, "%d-%b-%Y")
+    except ValueError:
+        yield line_num, f"{prefix} must be a 'DD-mmm-YYYY' date: {date_str}"
+        return
+
+    if dt.year < 1990:
+        yield line_num, f"{prefix} must not be before Python was invented: {date_str}"
+    if dt > (datetime.datetime.now() + datetime.timedelta(days=14)):
+        yield line_num, f"{prefix} must not be in the future: {date_str}"
+
+
+if __name__ == '__main__':
```

Comment:
To match the rest of the file:
```suggestion
if __name__ == "__main__":
```